starknet_py/net/gateway_client.py

```diff
@@ -0,0 +1,328 @@
+import typing
+import asyncio
+from typing import Union, Optional, List
+
+import aiohttp
+from marshmallow import EXCLUDE
+
+from starkware.starknet.definitions.fields import ContractAddressSalt
+from starkware.starkware_utils.error_handling import StarkErrorCode
+
+
+from starknet_py.transaction_exceptions import (
+    TransactionFailedError,
+    TransactionRejectedError,
+    TransactionNotReceivedError,
+)
+from starknet_py.net.base_client import (
+    BaseClient,
+    BlockHashIdentifier,
+    BlockNumberIdentifier,
+)
+from starknet_py.net.client_models import (
+    Transaction,
+    SentTransaction,
+    ContractCode,
+    TransactionReceipt,
+    BlockState,
+    StarknetBlock,
+    InvokeFunction,
+    StarknetTransaction,
+    ContractDefinition,
+    Deploy,
+    TransactionStatus,
+)
+from starknet_py.net.gateway_schemas.gateway_schemas import (
+    TransactionSchema,
+    ContractCodeSchema,
+    StarknetBlockSchema,
+    TransactionReceiptSchema,
+    SentTransactionSchema,
+)
+from starknet_py.net.models import StarknetChainId, chain_from_network
+from starknet_py.net.networks import Network, net_address_from_net
+
+
+class GatewayClient(BaseClient):
+    def __init__(self, net: Network, chain: StarknetChainId = None):
```

Comment:
Original Client included parameter `retry_config` as required by cairo-lang gateway client. Should we introduce a similiar parameter?

Reply:
I think it is pointless, retries should be handled on user's side.